cmd/updateIntegrationArtifactConfiguration_test.go

```diff
@@ -0,0 +1,90 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type updateIntegrationArtifactConfigurationMockUtils struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func newUpdateIntegrationArtifactConfigurationTestsUtils() updateIntegrationArtifactConfigurationMockUtils {
+	utils := updateIntegrationArtifactConfigurationMockUtils{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return utils
+}
+
+func TestRunUpdateIntegrationArtifactConfiguration(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Successfully update of Integration Flow configuration parameter test", func(t *testing.T) {
+		config := updateIntegrationArtifactConfigurationOptions{
+			Host:                   "https://demo",
+			OAuthTokenProviderURL:  "https://demo/oauth/token",
+			Username:               "demouser",
+			Password:               "******",
+			IntegrationFlowID:      "flow1",
+			IntegrationFlowVersion: "1.0.1",
+			ParameterKey:           "myheader",
+			ParameterValue:         "def",
+		}
+
+		httpClient := httpMockCpis{CPIFunction: "UpdateIntegrationArtifactConfiguration", ResponseBody: ``, TestType: "Positive", Method: "PUT", URL: "https://demo/api/v1/IntegrationDesigntimeArtifacts(Id='flow1',Version='1.0.1')"}
+
+		err := runUpdateIntegrationArtifactConfiguration(&config, nil, &httpClient)
+
+		if assert.NoError(t, err) {
+
+			t.Run("check url", func(t *testing.T) {
```

Comment:
What is the reason of encapsulating the assert in another test run?

Reply:
... I have the same habit. In case there is an error it does not make sense to continue with the remaining tests. In contrast to junit the go testing framework continues after a failed assert.

Reply:
I did not know that. So like this it stops the entire test here on failure.

Reply:
> So like this it stops the entire test here on failure.

it does not execute the tests in the `if` block in case the assert in the `if` condition fails ...

Reply:
Ah, yes, that was clear to me and I do the same. The comment above concerns the t.Run statement on line 44.